resolwe_bio/processes/reads_processing/markduplicates.py

```diff
@@ -0,0 +1,174 @@
+        validation_stringency = StringField(
+            label='Validation stringency',
+            description='Validation stringency for all SAM files read by this '
+                        'program. Setting stringency to SILENT can improve '
+                        'performance when processing a BAM file in which '
+                        'variable-length data (read, qualities, tags) do not '
+                        'otherwise need to be decoded. Default is STRICT.',
+            default='STRICT',
+            required=False
+        )
+        assume_sort_order = StringField(
+            label='Assume sort order',
+            description='If not null (default), assume that the input file '
+                        'has this order even if the header says otherwise.'
+                        'Possible values are unsorted, queryname, coordinate '
+                        'and unknown.',
+            default='',
+            required=False
+        )
+
+    class Output:
+        """Output fields to process MarkDuplicates."""
+        bam = FileField(label='Marked duplicates BAM file')
+        bai = FileField(label='Index of marked duplicates BAM file')
+        stats = FileField(label='Alignment statistics')
+        bigwig = FileField(label='BigWig file')
+        species = StringField(label='Species')
+        build = StringField(label='Build')
+        metrics_file = FileField(label='Metrics from MarkDuplicate process')
+
+    def run(self, inputs, outputs):
+        """
+        Note that this process can have output as two different filenames. If the process
+        is skipped, there will be no modification of the filename, whereas if
+        markduplication goes through, it will append 'markduplicates'.
+        """
+        # Prepare output file names.
+        file_name = os.path.basename(inputs.bam.bam.path)[:-4]
+        metrics_file = f'{file_name}_metrics.txt'
+        # We do not append anything particular to this object (like we did with e.g.
+        # _metrics.txt" for metrics file), because this step can be skipped and the
+        # name (e.g. "file_markduplicated.bam") would only confuse the matter.
+
+        species = inputs.bam.species
+        build = inputs.bam.build
+
+        if not inputs.skip:
+            if inputs.remove_duplicates:
+                rmd = 'true'
+            else:
+                rmd = 'false'
+
+            bam = file_name + '.markduplicates.bam'
+            md_inputs = [
+                '--INPUT', f'{inputs.bam.bam.path}',
+                '--VALIDATION_STRINGENCY', f'{inputs.validation_stringency}',
+                '--OUTPUT', f'{bam}',
+                '--METRICS_FILE', f'{metrics_file}']
+
+            if inputs.remove_duplicates:
+                md_inputs.extend(['--REMOVE_DUPLICATES', f'{rmd}'])
+
+            if inputs.assume_sort_order is not '':
+                md_inputs.extend(['--ASSUME_SORT_ORDER', f'{inputs.input.assume_sort_order}'])
+
+            print(md_inputs)
```

Comment:
Remove this print. Parameters used are listed in the `Cmd` stdout

Reply:
Remnants of my debugging attempts, sorry.